### PR TITLE
Legacy images

### PR DESCRIPTION
**Pull request by brianlovin**
**Status**
- [ ] WIP
- [x] Ready for review
- [ ] Needs testing

**Deploy after merge (delete what needn't be deployed)**
- api
- hyperion (frontend)

**Run database migrations (delete if no migration was added)**
YES

### CODE REVIEW

**Comment by brianlovin** [comment]
Failed `test_web` can be safely ignored: e2e passes, this seems to be a hiccup in the desktop test process although those tests normally pass (https://github.com/withspectrum/spectrum/pull/3645) - rerunning build to try again

File: api/migrations/20181102040518-fix-old-image-urls-in-threads.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
+exports.up = async function(r, conn) {
+  const threads = await r
+    .db('spectrum')
+    .table('threads')
+    .filter(row => row('createdAt').lt(r.epochTime(1540929600)))
```
Comment (mxstbr) [inline api/migrations/20181102040518-fix-old-image-urls-in-threads.js]: Should this check for `editedAt`?

File: api/migrations/20181102040518-fix-old-image-urls-in-threads.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    const processImageUrl = str => {
+      if (str.indexOf(LEGACY_PREFIX) < 0) {
+        return str;
+      }
+      if (str.indexOf('%20') < 0) {
+        return str;
+      }
+
+      const split = str.split('?');
```
Comment (mxstbr) [inline api/migrations/20181102040518-fix-old-image-urls-in-threads.js:32]: This seems very finnicky, will this break images that have `?` in the filename? Is there a safer way to do this?

If this is a URL and you want to get the search, maybe we can parse the url with the Node core `import { URL } from 'url'` module?

**Review by mxstbr — DISMISSED** [review]
Overall seems very good, a couple notes inline but other than that if it works, ship it!

File: api/migrations/20181102040518-fix-old-image-urls-in-threads.js (hunk 2 above)
Comment (brianlovin) [reply api/migrations/20181102040518-fix-old-image-urls-in-threads.js:32]: This works for now, let's run it. If this is too finnicky then we also need to edit how I've written the `editThread` and `publishThread` mutations :)

**Comment by brianlovin** [comment]
I'm going to do one defensive thing here and add extra fields to threads/messages/communities/users that get affected these migrations, just in case.

**Comment by brianlovin** [comment]
Migrations run in prod. Everything is looking solid. Will merge the additional fixes in this branch and create another migration in a few days to tear down those extra fields I added

**Pull request merged**